src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -138,7 +138,9 @@ else if (parseResult.isUsageHelpRequested()) {
                 final LocalizedMessage errorCounterMessage = new LocalizedMessage(1,
                         Definitions.CHECKSTYLE_BUNDLE, ERROR_COUNTER,
                         new String[] {String.valueOf(errorCounter)}, null, Main.class, null);
-                System.out.println(errorCounterMessage.getMessage());
+                // print error count statistic to error output stream,
+                // output stream might be used by validation report content
+                System.err.println(errorCounterMessage.getMessage());
```

Comment:
I would suggest not moving this message to stderr, but instead ignore it for -g (or may be some other modes), e.g. in line 137 make if condition look like
`
if (shouldPrintResult(errorCounter, cliOptions)) {
`
and method
```
private static boolean shouldPrintResult(int errorCounter, CliOptions cliOptions) {
   return errorCounter > 0 && !cliOptions.generateXpathSuppressionsFile; // some other options can be added later
}
```

Reply:
@strkkk 
> I would suggest not moving this message to stderr, but instead ignore it for -g

What is wrong with stderr?

Reply:
In my opinion stderr is supposed to be used for error messages, but this is not error message, just some information about results.

Also, in rare case if someone used checkstyle cli in bash script, printing such message in stderr would change its logic.

Reply:
Let's not forget that "error" in context of checkstyle is severity of violation. Violations should be part of report (standard output).
Exception stacktraces are edge case, we print it in stderr, as it is failure of checkstyle generate report.
Questionable line of output is not part of any report, so it is not stdout. It only appears when error violations happened or exception.
Ideally we should print statistics of EACH violation severity in report (plain, xml) - it is separate issue.
So only-error line is weird a bit, or incomplete a bit. So if we move it out stdout, it is good by all means. 

For now I suggest just move it to stderr, so if somebody rely on it he can find it easily.

Other option is simply remove this line.

Reply:
@strkkk , please give me approval.

Reply:
This solution wont work in case user want to see file in output, e.g.
```
java -jar checkstyle.jar -c conf -g Class.java
```
will produce same line in output in console even if you will write it into stderr.

Why don't you want to disable it for some work modes such as -g, as I suggested in initial comment?

Reply:
because it problem is not only for `-g`, take a look at #6986 .
Author of issue #6974 also suggested to move statistic to stderr, to have options to skip if redirecting to file only stdout content (ignoring or directing stderr to another file).

Complete removal might be an option, but users might already adjusted their scripts for this already, so we need to give them something back instead, it will be done at #6986 .